test/holidays/definition/repository/test_cache.rb

```diff
@@ -7,20 +7,55 @@ def setup
     @subject = Holidays::Definition::Repository::Cache.new
   end
 
+  def test_find_supports_overlapping_holidays
```

Comment:
I'm not sure if this test case is valuable in the real world, but felt obliged due to the semantics of the data structures.